Library/Homebrew/dev-cmd/bottle.rb

```diff
@@ -276,6 +277,7 @@ def formula_ignores(f)
     # https://gcc-help.gcc.gnu.narkive.com/GnwuCA7l/moving-gcc-from-the-installation-path-is-it-allowed.
     when Version.formula_optionally_versioned_regex(:gcc)
       %r{#{cellar_regex}/gcc} if OS.linux?
+      %r{#{prefix_regex}/opt/gcc} if OS.linux?
```

Comment:
This won't work because it's only going to return the latter.

Reply:
Ah right this needs to have an "or" operation combining both regexes.  Would `%r{#{cellar_regex}/gcc|#{prefix_regex}/opt/gcc}` work here?  I'll have to test this in Linux CI but should probably work out the right regex syntax first.

Reply:
`Regexp.union` is probably what you want

Reply:
This is indeed what was needed for both regexes to work.  I've now tested this with an older version of GCC that has only cellar references and GCC 12, which has the opt dir references, and both bottles end up with an `:any` cellar.  I think this will be ready to go now.